Add docs for new iteration of ent operator import feature (#29406)
hashicorp/vault: website/content/docs/import/vault.mdx

---
layout: docs
page_title: Vault secret import source
description: The Vault source imports secrets from Vault to Vault.
---

# Vault secret import source

Use the Vault source to import secret data from a Vault cluster into your Vault instance.

## Argument reference

Refer to the [HCL syntax](/vault/docs/import#hcl-syntax-1) for arguments common to all source types.

## Additional arguments

- `address` `(string: required)` - API address of source vault cluster.
- `namespace` `(string: "")` - Optional namespace to read from.
- `mount` `(string: required)` - KV-v2 mount to read from.

> - `credentials_file` `(string: "")` - Path to a file containing a vault token.  Note that the environment
> variable VAULT_TOKEN can't normally be used for this, since the CLI will use VAULT_TOKEN to authenticate to
> the destination Vault.  Exception: it's possible to import from/to the same Vault cluster.

Reviewer: ```suggestion
- `credentials_file` `(string: "")` - Path to a file containing a vault token.
  You must provide `credentials_file` except when importing from/to the same
  Vault cluster because the CLI only uses `VAULT_TOKEN` to authenticate to the
  **destination** Vault instance.
```
Style correction: write in the active voice

Author: This rewrite loses a bit of fidelity.  It's not true that the CLI only uses VAULT_TOKEN for destinations; it will use them for sources as well.  But usually that's not what the user will want, because importing from and to the same vault cluster (e.g. from one namespace or mount to another) isn't a common use case.  I'll rework it to be active voice.

Author: Attempted rewrite in 6f7c69cb1a43583263b6691ee1631c8e03ef9575
